src/components/mixins/UtilityMixin.vue

```diff
@@ -37,9 +37,10 @@ export default class UtilityMixin extends Vue {
             try {
                 const game = this.$store.state.activeGame;
                 const profile = this.$store.getters['profile/activeProfile'].asImmutableProfile();
-                const combo = throwForR2Error(await ThunderstoreCombo.fromProtocol(protocolUrl, game));
+                const combos = [throwForR2Error(await ThunderstoreCombo.fromProtocol(protocolUrl, game))];
+                const downloadId = await this.$store.dispatch('download/addDownload', combos);
 
-                await this.$store.dispatch('download/downloadAndInstallSpecific', {profile, combo});
+                this.$store.dispatch('download/downloadAndInstallCombos', {combos, profile, downloadId});
```

Comment:
Why remove the `await`? Wouldn't this suggest that `modList` below is loaded before the installation might have finished?

Reply:
I Added await to the usages.